components/dashboard/src/data/workspaces/listen-to-workspace-ws-messages.ts

```diff
@@ -4,47 +4,67 @@
+import { Disposable } from "@gitpod/gitpod-protocol";
 import { useQueryClient } from "@tanstack/react-query";
 import { useEffect } from "react";
-import { getGitpodService } from "../../service/service";
 import { getListWorkspacesQueryKey, ListWorkspacesQueryResult } from "./list-workspaces-query";
 import { useCurrentOrg } from "../organizations/orgs-query";
+import { workspaceClient } from "../../service/public-api";
+import { WatchWorkspaceStatusResponse } from "@gitpod/public-api/lib/gitpod/v1/workspace_pb";
 
 export const useListenToWorkspacesWSMessages = () => {
     const queryClient = useQueryClient();
     const organizationId = useCurrentOrg().data?.id;
 
     useEffect(() => {
-        const disposable = getGitpodService().registerClient({
-            onInstanceUpdate: (instance: WorkspaceInstance) => {
-                const queryKey = getListWorkspacesQueryKey(organizationId);
-                let foundWorkspaces = false;
+        const disposable = disposableWatchWorkspaceStatus(undefined, (status) => {
+            const queryKey = getListWorkspacesQueryKey(organizationId);
+            let foundWorkspaces = false;
 
-                // Update the workspace with the latest instance
-                queryClient.setQueryData<ListWorkspacesQueryResult>(queryKey, (oldWorkspacesData) => {
-                    return oldWorkspacesData?.map((info) => {
-                        if (info.workspace.id !== instance.workspaceId) {
-                            return info;
-                        }
-
-                        foundWorkspaces = true;
-                        return {
-                            ...info,
-                            latestInstance: instance,
-                        };
-                    });
+            // Update the workspace with the latest instance
+            queryClient.setQueryData<ListWorkspacesQueryResult>(queryKey, (oldWorkspacesData) => {
+                return oldWorkspacesData?.map((info) => {
+                    if (info.id !== status.workspaceId) {
+                        return info;
+                    }
+                    foundWorkspaces = true;
+                    info.status = status.status;
+                    return info;
                 });
+            });
 
-                if (!foundWorkspaces) {
-                    // If the instance was for a workspace we don't have, it should get returned w/ an updated query
-                    queryClient.invalidateQueries({ queryKey });
-                }
-            },
+            if (!foundWorkspaces) {
+                // If the instance was for a workspace we don't have, it should get returned w/ an updated query
+                queryClient.invalidateQueries({ queryKey });
+            }
         });
 
         return () => {
             disposable.dispose();
         };
     }, [organizationId, queryClient]);
 };
+
+export const disposableWatchWorkspaceStatus = (
```

Comment:
it does not look reliable, we need proper retry in place like in https://github.com/gitpod-io/gitpod/blob/ba8d3a4f1c245dc510d69ea6247c606150508d82/components/dashboard/src/service/public-api.ts#L188-L215